id-repository/credential-request-generator/src/main/java/io/mosip/credential/request/generator/service/impl/CredentialRequestServiceImpl.java

```diff
@@ -101,12 +101,14 @@ public class CredentialRequestServiceImpl implements CredentialRequestService {
 	public ResponseWrapper<CredentialIssueResponse> createCredentialIssuance(CredentialIssueRequestDto credentialIssueRequestDto) {
 		LOGGER.debug(IdRepoSecurityManager.getUser(), CREDENTIAL_SERVICE, CREATE_CREDENTIAL,
 				"started creating credential");
+		String requestId=null;
 		List<ServiceError> errorList = new ArrayList<>();
 		ResponseWrapper<CredentialIssueResponse> credentialIssueResponseWrapper = new ResponseWrapper<CredentialIssueResponse>();
 
 		CredentialIssueResponse credentialIssueResponse = null;
-		String requestId = utilities.generateId();
-
+		if(credentialIssueRequestDto.getAdditionalData()!=null) {
+			 requestId= credentialIssueRequestDto.getAdditionalData().containsKey("credential_id") ? credentialIssueRequestDto.getAdditionalData().get("credential_id").toString() : utilities.generateId();
+		}

```

Comment:
This has to be handled in the new endpoint with V2 in the URL where the credentialId will be taken in the request directly

Reply:
taken care